### PR TITLE
Simplify httplib2 refresh/retry logic

### PR DESCRIPTION
**Pull request by theacodes**

### CODE REVIEW

**Comment by dhermes** [comment]
Keeps Travis green?

**Comment by jonparrott** [comment]
Should be as green as a fresh meadow.

**Comment by jonparrott** [comment]
And it is. :)

File: oauth2client/transport/httplib2.py [hunk 1]
```diff
@@ -99,15 +102,13 @@ def new_request(uri, method='GET', body=None, headers=None,
...
             if body_stream_position is not None:
                 body.seek(body_stream_position)
 
-            resp, content = request(
-                orig_request_method, uri, method, body,
-                headers, redirections, connection_type)
+            return new_request(
+                uri, method, body=body, headers=None,
+                redirections=redirections, connection_type=connection_type,
+                _credential_refresh_attempt=_credential_refresh_attempt+1)
```
Comment (dhermes) [inline oauth2client/transport/httplib2.py:111]: Should be spaces around the plus sign.

Comment (dhermes) [reply oauth2client/transport/httplib2.py:111]: Put it in your next PR?

File: oauth2client/transport/httplib2.py [hunk 2]
```diff
@@ -99,15 +102,13 @@ def new_request(uri, method='GET', body=None, headers=None,
             # as b'Authorization'.
             headers.pop(b'Authorization', None)
 
-            credentials._before_request(orig_request_method, uri, headers)
-            headers = _helpers.clean_headers(headers)
```
Comment (dhermes) [inline oauth2client/transport/httplib2.py:103]: Gone why?

**Review by dhermes — APPROVED** [review]
LGTM

File: oauth2client/transport/httplib2.py (hunk 1 above)
Comment (theacodes) [reply oauth2client/transport/httplib2.py:111]: Yes, will do.

File: oauth2client/transport/httplib2.py (hunk 2 above)
Comment (theacodes) [reply oauth2client/transport/httplib2.py:103]: As discussed over chat, because recursion.

**Comment by jonparrott** [comment]
@nathanielmanistaatgoogle I'm moving quickly because I have a lot of these staged, if I merge and you have comment, I'll address them in the next PR. :)

**Pull request merged**